packages/device_info/android/src/main/java/io/flutter/plugins/deviceinfo/DeviceInfoPlugin.java

```diff
@@ -77,4 +78,25 @@ public void onMethodCall(MethodCall call, Result result) {
       result.notImplemented();
     }
   }
+
+  //simple emulator detection based on the flutter tools detection
+  //and a couple of legacy detection systems
+  private boolean isEmulator() {
```

Comment:
Can the logic below be moved into Dart code? We seem to have all the necessary fields.